src/overlays/actors/ovl_En_Test6/z_en_test6.c

```diff
@@ -35,50 +53,1278 @@ const ActorInit En_Test6_InitVars = {
+            func_800FD654(globalCtx, &D_80A94048, this->unk_15C);
+            func_800FD698(globalCtx, D_80A94054, D_80A94058, this->unk_15C);
+
+            if (this->unk_27A == 90) {
+                this->unk_282 = 0;
+                if (ENTEST6_GET(&this->actor) == ENTEST6_24) {
+                    this->unk_27C = 0x200;
+                    this->unk_150 = 0.0f;
+                    sp4C = -100.0f;
+                } else {
+                    this->unk_27C = 0x570;
+                    this->unk_150 = 110.0f;
+                    sp4C = 100.0f;
+                }
+                this->unk_14C = 1.0f;
+
+                for (i = 0; i < ARRAY_COUNT(this->unk_20C); i++) {
+                    this->unk_20C[i].x = player->actor.world.pos.x;
+                    this->unk_20C[i].y = player->actor.world.pos.y;
+                    this->unk_20C[i].z = player->actor.world.pos.z;
+                }
+
+                this->unk_254 = ZeldaArena_Malloc(sizeof(Vec3f) * 64);
+                if (this->unk_254 != NULL) {
+                    for (i = 0; i < ARRAY_COUNT(this->unk_254[0]); i++) {
+                        (*this->unk_254)[i].x = (((2.0f * Rand_ZeroOne()) - 1.0f) * 40.0f) + temp_s3->eye.x +
+                                                ((temp_s3->at.x - temp_s3->eye.x) * .2f);
+                        (*this->unk_254)[i].y = (((2.0f * Rand_ZeroOne()) - 1.0f) * 120.0f) + temp_s3->eye.y +
+                                                ((temp_s3->at.y - temp_s3->eye.y) * 0.2f) + sp4C;
+                        (*this->unk_254)[i].z = (((2.0f * Rand_ZeroOne()) - 1.0f) * 40.0f) + temp_s3->eye.z +
+                                                ((temp_s3->at.z - temp_s3->eye.z) * .2f);
+                    }
+                }
+                func_80A90C08(0x78);
+                Distortion_SetType(0x20);
+                Distortion_SetCountdown(0x50);
+                globalCtx->unk_18844 = 1;
+                this->unk_274 = 95;
+            }
+            break;
+
+        case 95:
+            if (this->unk_27A > 80) {
+                this->unk_282 += 25;
+            }
+
+            if (this->unk_27A < 20) {
+                this->unk_282 -= 25;
+            }
+
+            func_800FD59C(globalCtx, &D_80A9404C, this->unk_15C);
+            func_800FD5E0(globalCtx, &D_80A94050, this->unk_15C);
+            func_800FD654(globalCtx, &D_80A94048, this->unk_15C);
+            func_800FD698(globalCtx, D_80A94054 + this->unk_282, D_80A94058 + this->unk_282, this->unk_15C);
+
+            this->unk_278 -= this->unk_27C;
+            temp_s0 = this->unk_278;
+            if (ENTEST6_GET(&this->actor) == ENTEST6_24) {
+                this->unk_27C += 8;
+                this->unk_150 += this->unk_14C;
+            } else {
+                this->unk_27C -= 8;
+                this->unk_150 -= this->unk_14C;
+            }
+
+            for (i = 0; i < ARRAY_COUNT(this->unk_20C); i++) {
+                temp_s0 += 0x2AAA;
```

Comment:
I think this would match too
```suggestion
                temp_s0 += 0x10000 / ARRAY_COUNT(this->unk_20C);
```